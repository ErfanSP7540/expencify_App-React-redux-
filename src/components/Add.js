import React from "react"
import Form from './ExpenseForm'
import { expensesAdd } from '../actions/expenses'
import { connect } from 'react-redux'


export class Add extends  React.Component{

    onSubmit = (input) => {
        this.props.submit(input)
        this.props.history.push('/');
    }

    render(){
        return (
            <div>
                <h1> this Add Page </h1>
                <Form onSubmit={ this.onSubmit }/>
            </div>
        )
    }
}


const mapDispatchToProps=(dispatch)=>
{
    return ({ submit: (input )=>dispatch(expensesAdd(input)) })
}

export default connect(undefined, mapDispatchToProps)(Add);
