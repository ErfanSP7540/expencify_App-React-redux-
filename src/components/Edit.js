import React from 'react';
import Form from './ExpenseForm'
import { expensesEdit, expensesRemove } from '../actions/expenses'
import { connect } from 'react-redux'
import moment from 'moment'


export class Edit extends React.Component{
    EditEvent = (input) => {
        this.props.editDispatch({ ...input, id: this.props.match.params.id })
        this.props.history.push('/');
    }
    RemoveEvent = () => {
        this.props.removeDispatch({ id: this.props.match.params.id })
        this.props.history.push('/');
    }

    render(){ return (

        <div> 
        Edit Page {this.props.match.params.id && ("ID = " + this.props.match.params.id) }
                <Form expense={{ ...this.props.expense, createAt: moment(this.props.expense.createAt)   }} 
                onSubmit={this.EditEvent} />
                <button onClick={this.RemoveEvent}> remove </button>
        </div>
    )}
}

const mapStateToProps = (state , props ) => {
    const   expense =  state.expenses.filter((expense) => { if (expense.id === props.match.params.id){return true} } )
    if(expense[0]){
     return ({ expense: expense[0] })       
    }else{
        console.log(props.history)
        props.history.push('/')
        return ({})
    }
}
const mapDispatchToprops = (dispatch)=>{
    return ({ 
        editDispatch    : (input) =>   dispatch(expensesEdit(input)),
        removeDispatch  : (input)    =>   dispatch(expensesRemove(input))
      })
}
export default connect(mapStateToProps,mapDispatchToprops)(Edit)