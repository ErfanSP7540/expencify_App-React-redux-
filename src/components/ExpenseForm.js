import React from 'react'
import moment from 'moment'
import 'react-dates/initialize';
import { DateRangePicker, SingleDatePicker, DayPickerRangeController } from 'react-dates';




const now = moment();

class Form extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            description: props.expense ? props.expense.description : '',
            amount: props.expense ? props.expense.amount :0,
            note: props.expense ? props.expense.note: 'this is a note',
            createAt: props.expense ? props.expense.createAt: moment(),
            datePickerFocus:false,
            error:''        
        }
    }
    handleDescription=(e)=>{
        
        const Cstate =  { error:'', description: e.target.value };
        this.setState(Cstate);
        //console.log(this.state.description);
    }

    handleAmount=(e)=>{
        
        if( e.target.value.match(/^\d*(\.\d{0,2})?$/)) {
            const Cstate = { error: '',amount: e.target.value };
            this.setState(Cstate);
        }   
    }
    handleNote=(e)=>{
        

        const Cstate = { note: e.target.value };
        this.setState(Cstate);
    }
    onSubmit =(e)=>{
        e.preventDefault();
        
        if(!this.state.description  ||  this.state.amount==0){
            this.setState({error:'please enter description and amount for your expenses'})
        }
        else
        {
            this.props.onSubmit({
                description: this.state.description,
                amount: this.state.amount,
                note: this.state.note,
                createAt: this.state.createAt.valueOf()});

            this.setState({
                description: '',
                amount: 0,
                note: 'this is a note',
                createAt: moment(),
                datePickerFocus: false,
                error: ''
            })
        }
    }


    
    render(){
        return (
            <div>
            {this.state.error && <p>{this.state.error}</p>}
            <form onSubmit={this.onSubmit}>
            <input  type="text" placeholder="Description" value={this.state.description} 
                        onChange={this.handleDescription}/>
            <input type="text" value={this.state.amount} 
                onChange = {this.handleAmount}/>
             <br/>
            <textarea name="message" rows="10" value={this.state.note} 
            onChange={this.handleNote} />
            <button onClick={this.onSubmit}> Add Expenses </button>
                

                <SingleDatePicker
                    date={this.state.createAt} // momentPropTypes.momentObj or null
                    onDateChange={date => this.setState({ createAt: date })} // PropTypes.func.isRequired
                    focused={this.state.datePickerFocus} // PropTypes.bool
                    onFocusChange={( focused ) => this.setState({ datePickerFocus:!!focused })} // PropTypes.func.isRequired
                    numberOfMonths={1}
                    isOutsideRange={(day)=>{return false;}}
                    > </SingleDatePicker>

            </form>
            </div>
        )
    }

  


}
SingleDatePicker.displayName = 'SingleDatePicker';


export default Form;