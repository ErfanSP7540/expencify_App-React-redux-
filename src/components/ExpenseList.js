import React from 'react'
import {connect} from 'react-redux'
import Item from './ExpenseListItem'
import visibleExpenses from '../selectors/expenses.js'


export const ExpenseList =(props)=>{

    return ( props.selector.length==0? (<p>Not Found Any Expences</p> ):( 
    <div>
            <h1> Expense List {props.filter.text}</h1>
            {props.selector.map(item => <Item key={item.id} {...item }  /> )}
    </div>)
)
}


const mapStateToProp = (store) =>{
        return ({
            expenses: store.expenses,
            filter: store.filter,
            selector: visibleExpenses(store.expenses, store.filter )
        })
    }


const conectedExpenseList = connect(mapStateToProp)(ExpenseList)
export default conectedExpenseList