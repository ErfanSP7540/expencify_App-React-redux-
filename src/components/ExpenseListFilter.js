import React from 'react'
import {connect} from 'react-redux'
import { setTextFilter, setSortByDate, setSortByAmount, setStartTime, setEndTime } from '../actions/filters'

import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
//import 'react-dates/lib/css/_datepicker.css';

class ExpenseslistFilter extends  React.Component{
    constructor(props){
        super(props)
        
    this.state = {
            startDate: props.filter.startDate,
            endDate: props.filter.endDate,
        }
    }
    

render(){
    return (
            <div>
                <select value={this.props.filter.sortBy} onChange={(e)=>{
                    if( e.target.value==="amount")
                    {
                        this.props.dispatch(setSortByAmount());
                    }
                    else{
                        this.props.dispatch(setSortByDate());
                    }
                }}>
                    <option value="amount">amount</option>
                    <option value="date">date</option>
                </select> 
            <input value={this.props.filter.text} 
            onChange={(e)=>{
                this.props.dispatch(setTextFilter(e.target.value))
            }}/>



            <DateRangePicker
                startDate={this.state.startDate} // momentPropTypes.momentObj or null,
                startDateId="your_unique_start_date_id" // PropTypes.string.isRequired,
                endDate={this.state.endDate} // momentPropTypes.momentObj or null,
                endDateId="your_unique_end_date_id" // PropTypes.string.isRequired,
                onDatesChange={({ startDate, endDate }) => { 
                    this.setState({ startDate, endDate })
                    this.props.dispatch(setStartTime(startDate))
                    this.props.dispatch(setEndTime(endDate))
                }
                } // PropTypes.func.isRequired,
                focusedInput={this.state.focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                onFocusChange={focusedInput => this.setState({ focusedInput })} // PropTypes.func.isRequired,
            />


            </div>
    )
}
}


const mapStateToProps = (state)=>(
    { filter : state.filter  }
) 

export default connect(mapStateToProps)(ExpenseslistFilter) ;