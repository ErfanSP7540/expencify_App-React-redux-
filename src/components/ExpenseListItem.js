import React from 'react'
import { connect } from 'react-redux';
import { expensesRemove } from '../actions/expenses'
import { NavLink } from 'react-router-dom';
import moment from 'moment'
export const Item = ({dispatch , id , description, amount, note, createAt})=>(
    <div>
        <h3>

             <NavLink key={id} to={'/edit/' + id } ><button>Edit</button></NavLink>
              
             (description  : {description && description})
             (amount       : {amount && amount})          
             ( note         : {note && note})         
             (ceateAt      : {createAt && moment(createAt).format('YYYY MM DD')  })    
        </h3>
    </div>
)
export default  connect()( Item )