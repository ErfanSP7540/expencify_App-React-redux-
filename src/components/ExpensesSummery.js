import React from 'react'
import { connect } from 'react-redux'
import totalamount from '../selectors/expenses-total'
import numeral from 'numeral' //{amount && numeral(amount).format('$0,0.00')}) 
import visibleExpenses from '../selectors/expenses.js'


export class ExpensesSummery extends React.Component {

    constructor(props){
        super(props)
        console.log('summery2');
    }

    render(){
        
        return ( 
            <h2> Viewing {this.props.selector.length} expenses
             totalling {numeral(totalamount(this.props.selector)).format('$0,0.00')}
            </h2>
        )
    }

}


const mapStore2props=(store)=>{
    return { selector: visibleExpenses(store.expenses, store.filter)}
    
}

export default connect(mapStore2props)(ExpensesSummery);