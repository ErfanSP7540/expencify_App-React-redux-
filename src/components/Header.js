import React from 'react'
import {NavLink} from 'react-router-dom';

const Header =()=>(
    <div><h1>Header Text</h1>
        <NavLink to="/" exact> Home  </NavLink>
        <NavLink to="/Help" > Help  </NavLink>
        <NavLink to="/Add" > Add </NavLink>
        <NavLink to="/Edit" > Edit  </NavLink>
    </div>
)

export default Header;