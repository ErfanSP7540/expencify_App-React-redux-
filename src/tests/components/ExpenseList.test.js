import React from 'react'

import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


import { expenses , filter } from '../fixture-data'
import { ExpenseList} from '../../components/ExpenseList'




test('this is a fixture Data' , ()=>{

    const wrapper = shallow(<ExpenseList selector={expenses}  filter={filter}/>);
    expect(wrapper).toMatchSnapshot();

    
})

test('this is a fixture Data empty array', () => {

    const wrapper = shallow(<ExpenseList selector={[]} filter={filter} />);
    expect(wrapper).toMatchSnapshot();
})