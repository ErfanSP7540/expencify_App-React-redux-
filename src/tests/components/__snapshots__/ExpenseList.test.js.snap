// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this is a fixture Data 1`] = `
<div>
  <h1>
     Expense List 
    w
  </h1>
  <Connect(Item)
    amount={300}
    createAt={"1970-01-01T00:00:00.101Z"}
    description="water Bill"
    id="ididididididididi"
    key="ididididididididi"
    note="this is a Note(0)"
  />
  <Connect(Item)
    amount={200}
    createAt={"1970-01-01T00:00:00.100Z"}
    description="gas Bill"
    note="this is a Note(1)"
  />
  <Connect(Item)
    amount={100}
    createAt={"1970-01-01T00:00:00.500Z"}
    description="phone Bill"
    note="this is a Note(2)"
  />
  <Connect(Item)
    amount={700}
    createAt={"1970-01-01T00:00:00.300Z"}
    description="university"
    note="this is a Note(3)"
  />
  <Connect(Item)
    amount={600}
    createAt={"1970-01-01T00:00:00.240Z"}
    description="apartment"
    note="this is a Note(4)"
  />
  <Connect(Item)
    amount={500}
    createAt={"1970-01-01T00:00:00.450Z"}
    description="child fee"
    note="this is a Note(5)"
  />
  <Connect(Item)
    amount={20}
    createAt={"1970-01-01T00:00:00.070Z"}
    description="bread "
    note="this is a Note(6)"
  />
</div>
`;

exports[`this is a fixture Data empty array 1`] = `
<p>
  Not Found Any Expences
</p>
`;
